chapters/packages.tex

```diff
@@ -332,16 +332,18 @@ \section{Multilingual Descriptions}\label{multilingual-descriptions}
 
 The texts in following Modelica constructs should be translated:
 \begin{itemize}
-\item description strings of component declarations and classes
-\item strings in the following annotations:
-\begin{itemize}
-\item \lstinline!Text.string!, \lstinline!Text.textString!
-\item \lstinline!missingInnerMessage!, \lstinline!obsolete!, \lstinline!unassignedMessage!
-\item \lstinline!Dialog.group!, \lstinline!Dialog.tab!
-\item \lstinline!Dialog.loadSelector.caption!, \lstinline!Dialog.loadSelector.filter!, \lstinline!Dialog.saveSelector.caption!, \lstinline!Dialog.saveSelector.filter!
-\item \lstinline!Documentation.info!, \lstinline!Documentation.revisions!
-\item \lstinline!Figure.title!, \lstinline!Plot.title!, \lstinline!Curve.legend!
-\end{itemize}
+\item Description strings of component declarations and classes, see uses of \lstinline[language=grammar]!description-string! in \cref{modelica-concrete-syntax}.
```

Comment:
The current formulation doesn't make it clear whether all occurrences of _description-string_ should be translated.

For example, if the following is meant to be translated, I think it needs to be made clear that _every_ _description-string_ is subject to translation:
```
element-modification :
   name [ modification ] description-string
```

If not, we should be more formal about which ones to translate.

Reply:
If we want to clarify that it is restricted to the normal description we should just write "Description strings for declarations of components and classes." and I think that is the best.

I know that it is also possible to have description strings on: extends-clauses, connect-statements, other equations, modifiers.

To me the other description strings are more like structured _internal_ comments than _external_ descriptions for users of the class; so the description for the class tells you when to use the class and the description for a parameter-component is appropriate for the parameter-dialog; the description on an equation may explain its origin - like
```
equation
  p*V=n*R*T "Ideal Gas Law;
```

Reply:
According to https://github.com/modelica/ModelicaSpecification/issues/2237 description strings in modifiers override the  original ones, so I guess they are just as "external" as description strings directly on component declarations?

Reply:
The proposal would be simpler and better if we didn't make exceptions for any description strings.  I mean, why prevent that someone makes a translation of an equation's description?  The day some tool actually makes some use of these descriptions it would seem silly if they couldn't be translated.

Reply:
The problem with equation-descriptions is that we lack context for them, consider cases like:
```
model M
...
equation
...
  x=u1 "Using first input";
  y=u2 "Using other input";
end M;
```
It's also easier to later add more texts to list of things to translate than to remove them.

For modifiers there is at least a natural context.

Reply:
Why wouldn't `M` be a valid context for these equation descriptions?

Maybe we should ask ourselves why Modelica supports equation descriptions at all?  If we don't have an answer to that question, maybe we cannot expect to find the answer to why one should be able to provide translations.  Also, as long as the arguments we give against allowing translation are arguments that apply equally well to the untranslated equation descriptions, I think we should either get rid of the equation descriptions in the language, or allow them to be translated – having one but not the other wouldn't make sense.

Reply:
To me there is a difference between internal model documentation (for people trying to read and modify the model) and external model documentation (for people using the model).

The translations in MCP/0035 are primarily for the external model documentation.
Equations descriptions and comments are part of the internal model documentation.

For parameters the description is used in the parameter-dialog when using the model which makes it part of the external model documentation - and more importantly the texts are normally written with this in mind.

Reply:
> The translations in MCP/0035 are primarily for the external model documentation.

That might be the case, but why artificially restrict ourselves in a way that complicates the specification without adding any substantial value to the feature?

(I'm not sure what you mean by "comments"; I thought translating comments were never on the table?)

Reply:
> > The translations in MCP/0035 are primarily for the external model documentation.
> 
> That might be the case, but why artificially restrict ourselves in a way that complicates the specification without adding any substantial value to the feature?

I'm not saying that it would be a problem to also translate them, if the MCP allowed that; so I'm not restricting anything.

The reason is that, the proposed text doesn't say that they _can_ be translated, but that they _should_ - which adds a cost for handling those cases (for tools and translators) - that are both rare and largely irrelevant for the users of the library.

> (I'm not sure what you mean by "comments"; I thought translating comments were never on the table?)

I'm saying that descriptions on equations are in a similar category as "comments"; as part of the internal documentation.

Reply:
> The reason is that, the proposed text doesn't say that they _can_ be translated, but that they _should_ - which adds a cost for handling those cases (for tools and translators) - that are both rare and largely irrelevant for the users of the library.

Aha, that explains my confusion.  I never realized that the list was given in the spirit of _should_.  I'll start a separate thread about the strength of the formulation.